DiscordChatExporter.Domain/Discord/Models/Message.cs

```diff
@@ -90,6 +99,8 @@ public static Message Parse(JsonElement json)
             var callEndedTimestamp = json.GetPropertyOrNull("call")?.GetPropertyOrNull("ended_timestamp")?.GetDateTimeOffset();
             var type = (MessageType) json.GetProperty("type").GetInt32();
             var isPinned = json.GetPropertyOrNull("pinned")?.GetBoolean() ?? false;
+            var messageReference = json.GetPropertyOrNull("message_reference")?.Pipe(MessageReference.Parse);
+            var referencedMessage = json.GetPropertyOrNull("referenced_message")?.Pipe(Message.Parse);
```

Comment:
Do we need both `message_reference` and `referenced_message`?
From the docs, it seems that `referenced_message` will always be present in replies, unless the backend didn't fetch it, in which case we may as well pretend that there is no referenced message.

![image](https://user-images.githubusercontent.com/1935960/102694197-33326d80-4228-11eb-84b7-0ff21f2fe047.png)

Reply:
>in which case we may as well pretend that there is no referenced message.

Disagreed, if the message wasn't fetched (which could also happen if it was deleted), you still want to preserve the information that this message was a reply for context.  Otherwise you could get a discussion that reads in a misleading way.  Discord itself seem to convey it like this:
![image](https://user-images.githubusercontent.com/289437/102694612-7f7aaf00-4222-11eb-96ff-2f39489f861e.png)

Reply:
Ah, I see. I didn't account for deleted messages. Makes sense.